### PR TITLE
Uptake pekko jars, use pekko names in packages and configs

### PR DESCRIPTION
**Pull request by pjfanning**
* relates to https://github.com/apache/incubator-pekko-grpc/issues/21
* use pekko and pekko-http jars
* use pekko prefix in config names
* temporarily removed akka-http-cors in this repo because that lib does not yet support pekko - but soon will (#22)
* some pekko renaming
* have left the build of the doc updates until later (to keep down size of this PR)

### CODE REVIEW

**Comment by jrudolph** [comment]
Great stuff!

> * temporarily added akka-http-cors in this repo because that lib does not yet support pekko - but soon will

Isn't that a blocker for merging this PR for now? Feels wrong to add third-party code in such a big commit just to get everything working right now.

How about we just disable cors support for now to make this PR mergable (it's only used in four lines in the code) and then decide afterwards if we want to temporarily enable by importing the third-party project in its own PR or wait until the project itself has released something?

**Comment by jrudolph** [comment]
> > * temporarily added akka-http-cors in this repo because that lib does not yet support pekko - but soon will
> 
> Isn't that a blocker for merging this PR for now? Feels wrong to add third-party code as part of such a big commit just to get everything working right now.

If we would really want to add it to this project, we would have change its package name to avoid future problems for users which use both pekko-http-cors and pekko-grpc.

**Comment by pjfanning** [comment]
I guess I had the pekko version of akka-http-cors ready so it seemed easiest to use it but I can look at disabling cors support instead.

**Comment by mdedetrich** [comment]
Another option would be to wait a couple of weeks, the owner of `akka-http-cors` did say they are going to migrate over to pekko and since the `akka-http-cors` is pretty trivial it shouldn't be too problematic.

**Comment by pjfanning** [comment]
The cors code in pekko-grpc does look like it is non-trivial to rip out (and later add back) - so we might be best off waiting

**Comment by jrudolph** [comment]
> The cors code in pekko-grpc does look like it is non-trivial to rip out (and later add back) - so we might be best off waiting

Was easy enough, I added it here: https://github.com/pjfanning/incubator-pekko-grpc/pull/1. This might break something but it's only used in grpc-web which is less used than regular grpc support. I don't mind waiting for cors support, on the other hand, getting a snapshot version out which mostly works besides grpc-web/cors support might still be interesting.

**Comment by pjfanning** [comment]
I merged @jrudolph change. I'm happy to have this merged if the consensus is that we can look at adding back the CORS support in a few weeks.

**Comment by mdedetrich** [comment]
Fine with me, just make an issue for it so we don't forget

**Comment by pjfanning** [comment]
> * reintroduce dependency on akka-http-cors (or pekko equivalent)  #22

There is #22

File: docs/src/main/paradox/client/details.md
Comment (pjfanning) [reply docs/src/main/paradox/client/details.md:27]: #25 is open and probably can only be fixed as part of #26

Comment (mdedetrich) [reply docs/src/main/paradox/client/details.md:27]: I will just quickly verify whether `apidocRootPackage` is the cause of the problem

Comment (mdedetrich) [reply docs/src/main/paradox/client/details.md:27]: So I can verify if you modify `apidocRootPackage` on line 210 of `build.sbt` to `"org.apache.pekko"` then `@apidoc` works as expected (verified locally by doing `pekko-grpc-docs/paradox`). Also you need to change line 9 of `docs/src/main/paradox/deploy.md` from `@apidoc[akka.http.(javadsl|scaladsl).HttpsConnectionContext]` to `@apidoc[org.apache.pekko.http.(javadsl|scaladsl).HttpsConnectionContext]`

I think its easier if you revert that commit and apply this fix, alternately I can just add the commit to fix this onto this branch.

**Comment by pjfanning** [comment]
I've tried out the apidoc changes but CI is struggling due to Jabba connectivity issues

**Comment by mdedetrich** [comment]
Hmm, the jabba issues seem completely unrelated, jabba is used in `setup-java` github action just to download JDK. Let me see

**Comment by pjfanning** [comment]
Jabba will probably start working again. I can do a separate PR for setup-java switch. I can do that now.

**Comment by mdedetrich** [comment]
> Jabba will probably start working again. I can do a separate PR for setup-java switch. I can do that now.

I think it makes sense to do it now, its a trivial PR and needs to be done anyways (`setup-scala` as we can see has problems).

File: interop-tests/src/test/scala/org/apache/pekko/grpc/interop/GrpcInteropSpec.scala
Comment (mdedetrich) [inline interop-tests/src/test/scala/org/apache/pekko/grpc/interop/GrpcInteropSpec.scala:31]: I just noticed that this file still has clases/types with the `Akka` name in it, since the PR changes class names as well shall change it here as well?

**Review by mdedetrich — APPROVED** [review]
Okay I am going to approve this, tests are passing. I noticed some more cases where we are still using Akka in class/object names so feel free to update those before merging.

As mentioned before the only other thing that needs to be changed is to make the import style consistent with the Pekko akka projects, I can go ahead and do this after the PR is merged.

File: interop-tests/src/test/scala/org/apache/pekko/grpc/interop/GrpcInteropSpec.scala
Comment (pjfanning) [reply interop-tests/src/test/scala/org/apache/pekko/grpc/interop/GrpcInteropSpec.scala]: I updated this instance - there are still planty of Akka refs in docs, etc. - and it may be better to leave the other instances until another PR to avoid this one becoming too large

Comment (mdedetrich) [reply interop-tests/src/test/scala/org/apache/pekko/grpc/interop/GrpcInteropSpec.scala]: No worries, PR is approved anyways so its ready to merge when you are happy with it.

**Pull request merged**
